File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/MultiOnClickListener.java
package com.mapbox.services.android.navigation.ui.v5;

import android.view.View;

import java.util.HashSet;
import java.util.Set;

class MultiOnClickListener implements View.OnClickListener {
  private Set<View.OnClickListener> onClickListeners;

  MultiOnClickListener() {
    this.onClickListeners = new HashSet<>();
  }

  void addListener(View.OnClickListener onClickListener) {
    onClickListeners.add(onClickListener);
  }

  void removeListener(View.OnClickListener onClickListener) {
    onClickListeners.remove(onClickListener);
  }

  void clearListeners() {
    onClickListeners.clear();
  }

  @Override
  public void onClick(View view) {
    for (View.OnClickListener onClickListener : onClickListeners) {
      onClickListener.onClick(view);
    }
  }
}

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/recenterbutton/view/MultiOnClickListener.kt
package com.mapbox.navigation.ui.maps.recenterbutton.view

import android.view.View

internal class MultiOnClickListener : View.OnClickListener {

    private val onClickListeners: MutableSet<View.OnClickListener>

    fun addListener(onClickListener: View.OnClickListener) {
        onClickListeners.add(onClickListener)
    }

    fun removeListener(onClickListener: View.OnClickListener) {
        onClickListeners.remove(onClickListener)
    }

    fun clearListeners() {
        onClickListeners.clear()
    }

    override fun onClick(view: View) {
        for (onClickListener in onClickListeners) {
            onClickListener.onClick(view)
        }
    }

    init {
        onClickListeners = HashSet()
    }
}